apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/newchat/NewChatView.kt

```diff
@@ -0,0 +1,435 @@
+  val connReqInvitation by remember { derivedStateOf { chatModel.showingInvitation.value?.connReq ?: "" } }
+  val creatingConnReq = rememberSaveable { mutableStateOf(false) }
+  val pastedLink = rememberSaveable { mutableStateOf("") }
+  LaunchedEffect(selection.value) {
+    if (
+      selection.value == NewChatOption.INVITE
+      && connReqInvitation.isEmpty()
+      && contactConnection.value == null
+      && !creatingConnReq.value
+    ) {
+      createInvitation(rh?.remoteHostId, creatingConnReq, connReqInvitation, contactConnection)
+    }
+  }
+  DisposableEffect(Unit) {
+    onDispose {
+      /** When [AddContactLearnMore] is open, we don't need to drop [ChatModel.showingInvitation].
+       * Otherwise, it will be called here AFTER [AddContactLearnMore] is launched and will clear the value too soon.
+       * It will be dropped automatically when connection established or when user goes away from this screen.
+       * It applies only to Android because on Desktop center space will not be overlapped by [AddContactLearnMore]
+       **/
+      if (chatModel.showingInvitation.value != null && (!ModalManager.center.hasModalsOpen() || appPlatform.isDesktop)) {
+        val conn = contactConnection.value
+        if (chatModel.showingInvitation.value?.connChatUsed == false && conn != null) {
+          AlertManager.shared.showAlertDialog(
+            title = generalGetString(MR.strings.keep_unused_invitation_question),
+            text = generalGetString(MR.strings.you_can_view_invitation_link_again),
+            confirmText = generalGetString(MR.strings.delete_verb),
+            dismissText = generalGetString(MR.strings.keep_invitation_link),
+            destructive = true,
+            onConfirm = {
+              withBGApi {
+                val chatInfo = ChatInfo.ContactConnection(conn)
+                controller.deleteChat(Chat(remoteHostId = rh?.remoteHostId, chatInfo = chatInfo, chatItems = listOf()))
+                if (chatModel.chatId.value == chatInfo.id) {
+                  chatModel.chatId.value = null
+                  ModalManager.end.closeModals()
+                }
+              }
+            }
+          )
+        }
+        chatModel.showingInvitation.value = null
+      }
+    }
+  }
+  val tabTitles = NewChatOption.values().map {
+    when(it) {
+      NewChatOption.INVITE ->
+        stringResource(MR.strings.add_contact_tab)
+      NewChatOption.CONNECT ->
+        stringResource(MR.strings.connect_via_link)
+    }
+  }
+
+  Column(
+    Modifier.fillMaxSize(),
+  ) {
+    Box(contentAlignment = Alignment.Center) {
+      val bottomPadding = DEFAULT_PADDING
+      AppBarTitle(stringResource(MR.strings.new_chat), hostDevice(rh?.remoteHostId), bottomPadding = bottomPadding)
+      Column(Modifier.align(Alignment.CenterEnd).padding(bottom = bottomPadding, end = DEFAULT_PADDING)) {
+        AddContactLearnMoreButton()
```

Comment:
when an info page is closed, it always goes back to the view that was initially shown, not the view that was shown before opening info:

meaning if you enter via "Add contact", go to Connect via link tab, open info, then close info - New chat would show Add contact tab; and vice versa if you open via scan qr code button in search, go to Add contact tab, open then close info - New chat would show Connect via link tab.

Same goes for Incognito info when opening via qr code.